python/pyspark/pandas/data_type_ops/base.py

```diff
@@ -188,6 +188,12 @@ def _as_other_type(
     return index_ops._with_new_scol(scol, field=InternalField(dtype=dtype))
 
 
+def _sanitize_list_like(operand: Any) -> None:
```

Comment:
I wonder if we should add them actually since we're typing explicitly which enables static analysis. No strong opinion tho.

Reply:
Python's type annotations are just a hint for users, but they can still pass any objects, unlike Scala/Java.
I think type checking and showing a proper message are our responsibility.